### PR TITLE
CMake: Remove ASM-ZOS

### PR DESCRIPTION
**Pull request by dnakamura**
zos specific asm dialect serves no purpose.

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by dnakamura** [comment]
Jenkins build zos

**Comment by dnakamura** [comment]
Jenkins build zos

File: cmake/modules/platform/os/zos.cmake [hunk 1]
```diff
@@ -52,11 +52,6 @@ install(TARGETS omr_ascii omr_ebcdic
 )
 
 macro(omr_os_global_setup)
```
Comment (0xdaryl) [inline cmake/modules/platform/os/zos.cmake:54]: Please fix the copyright header in this file.

**Review by 0xdaryl — CHANGES_REQUESTED** [review]

File: cmake/modules/platform/os/zos.cmake (hunk 1 above)
Comment (0xdaryl) [reply cmake/modules/platform/os/zos.cmake:54]: Ping?  Should be an easy update and we can get this PR off the books.

Comment (dnakamura) [reply cmake/modules/platform/os/zos.cmake:54]: sorry. got buried in my inbox

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Previous CI passed and just a comment was updated.  Merging...

**Pull request merged**